scripts/affiliate_server.py

```diff
@@ -309,21 +364,40 @@ def GET(self, isbn: str) -> str:
                 {"error": "rejected_isbn", "isbn10": isbn10, "isbn13": isbn13}
             )
 
+        input = web.input(priority=False)
+        priority = (
+            Priority.HIGH if input.get("high_priority") == "true" else Priority.LOW
+        )
+
         # Cache lookup by isbn13. If there's a hit return the product to the caller
         if product := cache.memcache_cache.get(f'amazon_product_{isbn13}'):
-            return json.dumps({"status": "success", "hit": product})
+            return json.dumps(
+                {
+                    "status": "success",
+                    "hit": product,
+                }
+            )
 
         # Cache misses will be submitted to Amazon as ASINs (isbn10)
         if isbn10 not in web.amazon_queue.queue:
-            web.amazon_queue.put_nowait(isbn10)
+            isbn10_queue_item = PrioritizedISBN(isbn=isbn10, priority=priority)
+            web.amazon_queue.put_nowait(isbn10_queue_item)
 
         # Give us a snapshot over time of how many new isbns are currently queued
         stats.put(
             "ol.affiliate.amazon.currently_queued_isbns",
             web.amazon_queue.qsize(),
             rate=0.2,
         )
-        return json.dumps({"status": "submitted", "queue": web.amazon_queue.qsize()})
+
+        # Check the cache a few times for product data to return to the client,
+        # or otherwise return.
+        for _ in range(3):
```

Comment:
We may want to make this a named constant or perhaps have it be adjustable.

1) let's increase this to 5?

2) let's stats.incr any time this fails to see how often failure cases are happening and adjust the number accordingly